### ISSUE TITLE
AWS SNS delivery status support

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#515]
_This issue was originally opened by @pjungermann as hashicorp/terraform#11739. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Hej Terraform team,

It would be great to support the SNS delivery status logging. This was already asked for at issue hashicorp/terraform#5386, but got closed due to inactivity.

You can find an example for how to enable it at the console and via the Java API at
- https://aws.amazon.com/blogs/mobile/using-the-delivery-status-feature-of-amazon-sns/
- http://docs.aws.amazon.com/sns/latest/dg/msg-status-topics.html#msg-status-sdk

I think it would fit either as a new attribute to the "aws_sns_topic" resource or maybe even better as  a new:

```
resource "aws_sns_topic_delivery_status" "sqs_delivery_status" {
  topic_arn           = "${aws_sns_topic.my_sns_topic.arn}"
  endpoint            = "sqs" # application, http, lambda, sqs, * / all (default)
  success_sample_rate = 50 # default: 100
  success_role_arn    = "${aws_iam_role.cloud_watch_logging.arn}"
  failure_role_arn    = "${aws_iam_role.cloud_watch_logging.arn}"
}
```

**Comment by StephenWithPH** [comment]
It appears this work is already done, but got lost in the repo splitting. See https://github.com/hashicorp/terraform/pull/14101. 

@gazoakley ... are you willing to port this over?

**Comment by gazoakley** [comment]
The work is awaiting review in #2872 

Originally thumbed up in https://github.com/hashicorp/terraform/issues/11739 by:

- @thyrion
- @johnhampton
- @shinzui
- @alfredgamulo
- @ligustah
- @llange
- @richid 
- @pagemedias 
- @jonbrouse
- @nathan-muir
- @pygillier
- @timdonegan
- @dgolja
- @grimm26
- @operatingops
- @Cuneytt
- @saketmalpure

If any of you are still waiting on this, please feel free to thumbs up these:

- https://github.com/terraform-providers/terraform-provider-aws/issues/515#issue-235665136
- https://github.com/terraform-providers/terraform-provider-aws/pull/2872#issue-286197905

**Comment by bflad** [comment]
@gazoakley thanks so much for porting this over and helping direct everyone here 💯  I'll be reviewing #2872 shortly.

**Comment by bflad** [comment]
The new attributes for `aws_sns_topic` to support delivery status have been merged into master in #2872 and will be released in v1.9.0 of the AWS provider, expected the end of this week. Happy Terraform'ing! 🎉

**Comment by gazoakley** [comment]
Thanks @bflad 👍

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.9.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#190-february-09-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
